server/src/completion/YmlCompletionItemsProvider.ts

```diff
@@ -26,7 +26,23 @@ export class YmlCompletionItemsProvider {
     );
   }
 
-  public getOnlyCompilationItems() {
-      return this.completions.map((elem) => elem.completion);
+  public getAvailableCompletionItems(uri: string, offset: number) {
+    return this.completions
+      .map((elem) => {
+        if (elem.scopeEndOffset && elem.scopeStartOffset && elem.uri === uri) {
+          // We have the scope's positions set and are in the correct file.
+          if (offset < elem.scopeStartOffset || offset > elem.scopeEndOffset) {
+            // Outside of the scope.
+            return null;
+          } else {
+            return elem.completion;
+          }
+        } else {
+          // No information about the scope. The element is available everywhere.
+          return elem.completion;
+        }
+      })
+      // Remove null elements.
+      .filter((elem) => !!elem);
```

Comment:
Silly question: is there an equivalent of Java's flatMap to avoid having to manually remove stuff afterwards? Like, instead of sprouting `null` values, you could return empty streams. Dunno.

Reply:
This is experimental and doesn't seem to work :(